lib/portlayer/network/context.go

```diff
@@ -1269,6 +1269,18 @@ func (c *Context) DeleteScope(ctx context.Context, name string) error {
 		return fmt.Errorf("cannot remove builtin scope")
 	}
 
+	// remove dangling endpoints
+	if exec.Containers != nil {
+		for _, eps := range s.Endpoints() {
+			if exec.Containers.Container(eps.ID().String()) == nil {
```

Comment:
small nit: instead of calling `eps.ID().String()` 3 times here you could assign it a variable once and then just use that variable instead :) will make the code a little more friendly to read.